lib/utils/fields.js

```diff
@@ -40,14 +47,67 @@ export function fieldUnitBinTest(fields, col) {
   );
 }
 
-const DATA_ARRAY_TO_OBJECT_OPTION_DEFAULTS = {
-  strict: true,
-};
+// parse failure behaviors
+export const KEEP_ORIGINAL = 'KEEP_ORIGINAL';
+export const REMOVE = 'REMOVE';
+export const THROW_ERROR = 'THROW_ERROR';
+
+// date formats
+export const DATE_OBJECT = 'DATE_OBJECT';
+export const JSON_SCHEMA_FORMAT = 'JSON_SCHEMA_FORMAT';
+export const INPUT_FORMAT = 'INPUT_FORMAT';
+
 export function dataArrayToObject(dataArray, fields, options = {}) {
-  const { strict } = {
-    ...DATA_ARRAY_TO_OBJECT_OPTION_DEFAULTS,
+  const {
+    parseFailureBehavior,
+    dateBehavior,
+    dateFormat,
+    datetimeFormat,
+    timeFormat,
+  } = {
+    parseFailureBehavior: KEEP_ORIGINAL,
+    dateBehavior: DATE_OBJECT,
+    dateFormat: 'yyyy-MM-dd',
+    datetimeFormat: 'yyyy-MM-dd HH:mm',
+    timeFormat: 'HH:mm',
     ...options,
   };
+  const datatypes = new Datatypes({
+    dateFormat,
+    datetimeFormat,
+    timeFormat,
+  });
+  const formatDatesAndTimes = (dateObject, datatype) => {
+    if (!isDate(dateObject)) {
+      return dateObject;
+    }
+    if (dateBehavior === DATE_OBJECT) {
+      return dateObject;
+    }
+    if (dateBehavior === INPUT_FORMAT) {
```

Comment:
And I gather that the INPUT_FORMAT is all about the way date/time picker fields format and show dates/times in HandsonTable cells on screen, when they initially were read in from a json data file into javascript objects?!